MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -771,7 +772,17 @@ - (void)setupLogBuffer {
 
       // Create a buffer for the priority. Making use of `{}` as we're using C++11.
       for (NSUInteger i = 0; i < ms_crashes_log_buffer_size; i++) {
-        msCrashesLogBuffer[MSPriority(priority)][i] = MSCrashesBufferedLog{files[i]};
+
+        /**
+         * Some explanation why we are initializing the msCrashLogBuffer with `files[i], nil` instead of `files[i]`:
+         * Not assigning `nil` leads to potentially unpredictable runtime behavior if you have any code which assumes
+         * the second field is always some valid or `nil` value (since failing to explicitly initialize, it permits the
+         * compiler to leave garbage in that field - zero-init is only guaranteed for Objective-C objects and static
+         * data, it is especially NOT guaranteed for a stack temporary like here. Even if zero-init were guaranteed,
+         * this is [Objective-]C++ so it's not legal to borrow ObjC's "nil is always zero"-rule so technically it would
+         * still not be properly initialized despite being a trivial type.
+         */
+        msCrashesLogBuffer[MSPriority(priority)][i] = MSCrashesBufferedLog{files[i], nil};
```

Comment:
Please refresh your branch. This line has a conflict, `files[i]` -> `[files[i] path]`.